d2/src/main/java/com/linkedin/d2/xds/balancer/DualReadXdsLoadBalancerFactory.java

```diff
@@ -0,0 +1,31 @@
+package com.linkedin.d2.xds.balancer;
+
+import com.linkedin.d2.balancer.D2ClientConfig;
+import com.linkedin.d2.balancer.LoadBalancerWithFacilities;
+import com.linkedin.d2.balancer.LoadBalancerWithFacilitiesFactory;
+import com.linkedin.d2.balancer.dualread.DualReadLoadBalancer;
+import com.linkedin.d2.balancer.dualread.DualReadModeProvider;
+import java.util.concurrent.Executors;
+
+
+public class DualReadXdsLoadBalancerFactory implements LoadBalancerWithFacilitiesFactory
+{
+  private final LoadBalancerWithFacilitiesFactory _oldLbFactory;
+  private final LoadBalancerWithFacilitiesFactory _newLbFactory;
+  private final DualReadModeProvider _dualReadModeProvider;
+
+  public DualReadXdsLoadBalancerFactory(LoadBalancerWithFacilitiesFactory oldLbFactory,
+      LoadBalancerWithFacilitiesFactory newLbFactory, DualReadModeProvider dualReadModeProvider)
```

Comment:
Nit: for the context of this class, we can just create XdsLoadBalancerWithFacilitiesFactory as the newLbFactory and remove it from the constructor parameter?

Reply:
Good point!